arrow-fx-rx2/src/main/kotlin/arrow/fx/rx2/extensions/maybek.kt

```diff
@@ -203,9 +203,15 @@ interface MaybeKConcurrent : Concurrent<ForMaybeK>, MaybeKAsync {
         val sb = ReplaySubject.create<B>()
         val dda = fa.value().subscribe(sa::onNext, sa::onError)
         val ddb = fb.value().subscribe(sb::onNext, sb::onError)
+
+        val fiberSa = ReplaySubject.create<A>()
+        val fiberSb = ReplaySubject.create<B>()
+        val fiberDda = fa.value().subscribe(fiberSa::onNext, fiberSa::onError)
+        val fiberDdb = fb.value().subscribe(fiberSb::onNext, fiberSb::onError)
```

Comment:
I never used Rx2 so I'm just guessing here, trying to make tests pass first. It seems that the `sa`/`sb` are disposed too early, and the `parMap2` that calls `racePair` never ends (stuck on `Fiber.join()`).

Reply:
Still, you're now executing each Observable twice, because you subscribe multiple times to the same fa/fb, which is not desired. If removing this code causes the tests to fail then we will know the bug is here, but the bugfix will need to follow a different approach.

Reply:
Sorry, since I never used Rx2, I erroneously supposed it was only "listening" twice... anyway the problem is that function. I suspect that the problem is related to the fact that the `subscribe` here in `racePair` breaks the `firstElement` inside the Fiber.

Reply:
Ok. I've found the problem and pushed a better fix (maybe not definitive). The real cause is in the `emitter.onSuccess` call, that internally disposes the emitter before returning. Since the `setCancellable` call disposes both the subjects, the second one couldn't never provide its value. I've tried different solutions but the only working one I found is to have a "configurable" disposal function that disposes only the subject that has already provided its value. The other one will be disposed when the corresponding fiber is cancelled.

If this is the right way to fix the problem, we should port it also to the `raceTriple` implementations.

Reply:
@colomboe looks like the correct solution to me 👍

Reply:
Thank you @nomisRev! I've applied the same fix also on `raceTriple` for both `SingleK` and `MaybeK`.